telethon/client/telegrambaseclient.py

```diff
@@ -654,6 +654,21 @@ def set_proxy(self: 'TelegramClient', proxy: typing.Union[tuple, dict]):
             else:
                 connection._proxy = proxy
 
+    def _save_states_and_entities(self: 'TelegramClient'):
+        entities = self._mb_entity_cache.get_all_entities()
+
+        # Piggy-back on an arbitrary TL type with users and chats so the session can understand to read the entities.
+        # It doesn't matter if we put users in the list of chats.
+        self.session.process_entities(types.contacts.ResolvedPeer(None, [e._as_input_peer() for e in entities], []))
+
+        ss, cs = self._message_box.session_state()
+        self.session.set_update_state(0, types.updates.State(**ss, unread_count=0))
+        now = datetime.datetime.now()  # any datetime works; channels don't need it
+        entities_ids = (e.id for e in entities)
```

Comment:
You changed `[]` for `()`, which made it a (lazy) generator, so this won't work as is. [What I suggested was braces](https://github.com/LonamiWebs/Telethon/pull/4071#discussion_r1150824548) (`{` and `}`), which will correctly make a `set`.

```suggestion
        entities_ids = {e.id for e in entities}
```

Reply:
Reread the documentation. Discovered that {} - dict, {1,2,3} - set. I did not encounter such subtlety. Now I will correct, sorry

Reply:
The first time I thought you made a mistake, but it turns out I don't have enough experience

Reply:
That's fine. `{}` being used for both `dict` and `set` is indeed confusing, but I'm glad you've learnt something new.